plugins/cost-insights/src/components/CostOverviewCard/CostOverviewByProductChart.tsx

```diff
@@ -0,0 +1,211 @@
+import { getPreviousPeriodTotalCost } from '../../utils/change';
+import { formatPeriod } from '../../utils/formatters';
+import { aggregationSum } from '../../utils/sum';
+import { BarChartLegendOptions } from '../BarChart/BarChartLegend';
+
+dayjs.extend(utc);
+
+export type CostOverviewByProductChartProps = {
+  costsByProduct: Cost[];
+};
+
+const LOW_COST_THRESHOLD = 0.1;
+
+export const CostOverviewByProductChart = ({
+  costsByProduct,
+}: CostOverviewByProductChartProps) => {
+  const theme = useTheme<CostInsightsTheme>();
+  const styles = useStyles(theme);
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const { duration } = useFilters(mapFiltersToProps);
+
+  if (!costsByProduct) {
+    return null;
+  }
+
+  const flattenedAggregation = costsByProduct
+    .map(cost => cost.aggregation)
+    .flat();
+
+  const totalCost = aggregationSum(flattenedAggregation);
+
+  const previousPeriodTotal = getPreviousPeriodTotalCost(
+    flattenedAggregation,
+    duration,
+    lastCompleteBillingDate,
+  );
+  const currentPeriodTotal = totalCost - previousPeriodTotal;
+  const productsByDate = costsByProduct.reduce((prodByDate, product) => {
+    const productTotal = aggregationSum(product.aggregation);
+    // Group products with less than 10% of the total cost into "Other" category
+    // when there we have >= 5 products.
+    const isOtherProduct =
+      costsByProduct.length >= 5 &&
+      productTotal < totalCost * LOW_COST_THRESHOLD;
+    const productName = isOtherProduct ? 'Other' : product.id;
+    const updatedProdByDate = { ...prodByDate };
+
+    product.aggregation.forEach(curAggregation => {
+      const productCostsForDate = updatedProdByDate[curAggregation.date] || {};
+
+      updatedProdByDate[curAggregation.date] = {
+        ...productCostsForDate,
+        [productName]:
+          (productCostsForDate[productName] || 0) + curAggregation.amount,
+      };
+    });
+
+    return updatedProdByDate;
+  }, {} as Record<string, Record<string, number>>);
+
+  const chartData: Record<string, number>[] = Object.keys(productsByDate).map(
+    date => {
+      return {
+        ...productsByDate[date],
+        date: Date.parse(date),
+      };
+    },
+  );
+
+  const renderAreas = () => {
+    const productGroupNames = new Set(
+      Object.values(productsByDate)
+        .map(d => Object.keys(d))
+        .flat(),
+    );
+    const sortedProducts = costsByProduct
+      // Check that product is a separate group and hasn't been added to 'Other'
+      .filter(
+        product => product.id !== 'Other' && productGroupNames.has(product.id),
+      )
+      .sort(
+        (a, b) => aggregationSum(a.aggregation) - aggregationSum(b.aggregation),
+      )
+      .map(product => product.id);
+    // Keep 'Other' category at the bottom of the stack
+    return ['Other', ...sortedProducts].map((product, i) => (
+      <Area
+        dataKey={product}
+        stackId="1"
+        fillOpacity="1"
```

Comment:
The colors seem suuuper bold, can we drop the fillOpacity? See where we go from face-punching colors to something still bold but more palatable.